### PR TITLE
Update ESLint dependencies and config

### PR DESCRIPTION
**Pull request by Turbo87**
... and fix the few additional warnings/errors that this update has caused

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,7 +1,83 @@
...
-    'simplabs/plugins/ember',
+    'eslint:recommended',
+    'plugin:ember/recommended'
   ],
+  env: {
+    browser: true
+  },
+  rules: {
+    // TODO this should no longer be necessary with our Ember support matrix
+    'ember/no-get': 'off',
```
Comment (marcoow) [inline .eslintrc.js:21]: Can this be removed then?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -49,6 +49,8 @@ jobs:
 
       - run: yarn install
 
+      - run: yarn lint:js
```
Comment (marcoow) [inline .github/workflows/ci.yml:52]: Can we run this once instead of for every supported Ember version?

File: .eslintrc.js (hunk 1 above)
Comment (Turbo87) [reply .eslintrc.js:21]: only once we remove the problematic parts of the code, but I wanted to do that in a dedicated PR

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Turbo87) [reply .github/workflows/ci.yml:52]: it's matching what we were already doing before, but generally we could, yes. I just wanted to keep the PR relatively small for now.

Comment (marcoow) [reply .github/workflows/ci.yml:52]: It seems before this we weren't running `lint` at all on CI though?

File: .eslintrc.js (hunk 1 above)
Comment (marcoow) [reply .eslintrc.js:21]: ok

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Turbo87) [reply .github/workflows/ci.yml:52]: we were running it via ember-cli-eslint, which generates Mocha tests that were run for each ember-try scenario

Comment (marcoow) [reply .github/workflows/ci.yml:52]: ah I see – ok 👍

**Review by marcoow — APPROVED** [review]

**Pull request merged**